llvm/lib/Transforms/Scalar/DeadStoreElimination.cpp

```diff
@@ -2261,6 +2261,19 @@ struct DSEState {
   bool eliminateDeadDefs(const MemoryDefWrapper &KillingDefWrapper);
 };
 
+// Return true if "Arg" is function local and isn't captured before "CB" or
+// if "Arg" is GEP whose base pointer is function local and isn't captured
+// before "CB".
+bool IsFuncLocalAndNotCaptured(Value *Arg, const CallBase *CB,
+                               EarliestEscapeAnalysis &EA) {
+  if (isIdentifiedFunctionLocal(Arg))
+    return EA.isNotCapturedBefore(Arg, CB, /*OrAt*/ true);
+  const auto *GEP = dyn_cast<GetElementPtrInst>(Arg);
```

Comment:
Check the result of getUnderlyingObject instead, it will look through multiple GEPs.

Reply:
Ah, done. Thanks!